File: Lista_6/lista6_ex6.c
/* Exercício 6:
Escreva uma função chamada media que recebe um vetor de double, um inteiro n que indica o tamanho do vetor, e um ponteiro para inteiro i. 
A função deve retornar a média dos n elementos no vetor e no endereço apontado por i deve retornar a posição do elemento que tem o valor mais próximo da média.
O protótipo da função deve ser:
double media(double vet[], int n, int *i);
*/
#include <stdio.h>

double media(double vet[], int n, int *i);

int main() {
int n, i;
double vet[100];

printf("Quantos numeros serao inseridos no vetor: ");
scanf("%d", &n);

printf("\nInsira os numeros em ordem crescente.\n");
for(int j=0; j < n; j++) {
    printf("\nInsira o %d elemento: ", j+1);
    scanf("%lf", &vet[j]);
}

printf("\nMedia: %.1lf", media(vet, n, &i));
printf("\nPosicao do valor mais proximo da media: %d\n", i);

return 0;
}

double media(double vet[], int n, int *i) {
double sum=0, mean, dmax, dmin;
int j, min, max;

    for(j=0; j < n; j++) {
        sum = sum + vet[j];
    }
    mean = sum/n;

    for(j=0; j < n; j++) {
        if(vet[j] > mean) {
            max = j;
            min = j-1;
            break;
        }
    }
    dmax = max - mean;
    dmin = mean - min;

    if(dmax > dmin) {
        *i = min;
    } else {
        *i = max;
    }
return mean;
}

File: Lista_6/Lista6_ex6.py
def cria_vetor(n):
	vetor = []
	for j in range(n):
		numero = float(input("\nInsira o %d elemento: " % (j)))
		vetor.append(numero)
	return vetor

def media(vetor):
	soma = 0
	n = len(vetor)

	for j in range(n):
		soma += vetor[j]
	m = soma / n

	aux = sorted(vetor) #organiza vetor em ordem crescente, transfere ordem para aux

	for j in range(n):
		if aux[j] > m:
			ma = aux[j]
			mi = aux[j-1]
			posma = j
			break
	
	dma = ma - m
	dmi = m - mi

	if dma > dmi:
		p = posma - 1
	else:
		p = posma
	
	for j in range(n): #compara vetores e obtem a posicao no vetor
		if(vetor[j] == aux[p]):
			i = j

	return m, i

def main():
	vetor = []
	
	n = int(input("Quantos numeros serao inseridos no vetor: "))

	vetor = cria_vetor(n)
	
	m, i = media(vetor)
	print("\nMedia: %.1lf" % m)
	print("\nPosicao do valor mais proximo da media: %d\n" % i)
	
if __name__ == "__main__":
	main()
